lib/jsdom/living/events/CustomEvent-impl.js

```diff
@@ -0,0 +1,26 @@
+"use strict";
+
+const EventImpl = require("./Event-impl").implementation;
+const CustomEventInit = require("../generated/CustomEventInit");
+
+class CustomEventImpl extends EventImpl {
+  constructor(args, privateData) {
+    const type = args[0]; // TODO: Replace with destructuring
+    const eventInitDict = args[1] || CustomEventInit.convert(undefined);
```

Comment:
We really need to fix this before this is merge-ready. It should not be necessary to define this constructor.